tufaceous-lib/src/artifact.rs

```diff
@@ -136,12 +138,29 @@ fn write_host_tarball_fake_artifact<W: Write>(
     size: u64,
     writer: &mut W,
 ) -> Result<()> {
-    let mut builder = tar::Builder::new(BufWriter::new(writer));
+    let mut builder = tar::Builder::new(flate2::write::GzEncoder::new(
+        BufWriter::new(writer),
+        Compression::fast(),
+    ));
 
     let times = (size as usize) / FILLER_TEXT.len();
     let phase_1_times = times / 8;
     let phase_2_times = times - phase_1_times;
 
+    {
+        let oxide_metadata_json =
+            serde_json::to_string(&OxideMetadata::os_metadata())
+                .expect("oxide metadata should serialize correctly");
+        let mut header = tar::Header::new_gnu();
```

Comment:
These should be `new_ustar()` not `new_gnu()` FWIW, and I think you probably need to `set_entry_type(tar::EntryType::Regular)` as well to make sure they're files in the archive.

Reply:
Thanks, updated.